File: example/src/androidTest/java/com/github/bassaer/example/util/ElapsedTimeIdlingResource.java
package com.github.bassaer.example.util;

import android.support.test.espresso.IdlingResource;

/**
 * Implement to wait for a specific amount of time
 * Created by nakayama on 2017/08/01.
 */

public class ElapsedTimeIdlingResource implements IdlingResource {
    private long mStartTime;
    private long mWaitingTime;
    private ResourceCallback mResourceCallback;

    public ElapsedTimeIdlingResource(long waitingTime) {
        mStartTime = System.currentTimeMillis();
        mWaitingTime = waitingTime;
    }
    @Override
    public String getName() {
        return ElapsedTimeIdlingResource.class.getName();
    }

    @Override
    public boolean isIdleNow() {
        long elapsed = System.currentTimeMillis() - mStartTime;
        boolean idle = elapsed >= mWaitingTime;
        if (idle) {
            mResourceCallback.onTransitionToIdle();
        }
        return idle;
    }

    @Override
    public void registerIdleTransitionCallback(ResourceCallback callback) {
        mResourceCallback = callback;
    }
}

File: example/src/androidTest/kotlin/com/github/bassaer/example/util/ElapsedTimeIdlingResource.kt
package com.github.bassaer.example.util

import androidx.test.espresso.IdlingResource

/**
 * Implement to wait for a specific amount of time
 * Created by nakayama on 2017/08/01.
 */
class ElapsedTimeIdlingResource(private val mWaitingTime: Long) : IdlingResource {
    private val mStartTime: Long = System.currentTimeMillis()
    private var mResourceCallback: IdlingResource.ResourceCallback? = null

    override fun getName(): String {
        return ElapsedTimeIdlingResource::class.java.name
    }

    override fun isIdleNow(): Boolean {
        val elapsed = System.currentTimeMillis() - mStartTime
        val idle = elapsed >= mWaitingTime
        if (idle) {
            mResourceCallback!!.onTransitionToIdle()
        }
        return idle
    }

    override fun registerIdleTransitionCallback(callback: IdlingResource.ResourceCallback) {
        mResourceCallback = callback
    }
}
